openvdb/openvdb/tree/InternalNode.h

```diff
@@ -268,6 +268,18 @@ class InternalNode
     /// Set the grid index coordinates of this node's local origin.
     void setOrigin(const Coord& origin) { mOrigin = origin; }
 
+#if OPENVDB_ABI_VERSION_NUMBER >= 9
+    /// Return the transient data value.
+    Index64 transientData() const { return Index64(mTransientData); }
```

Comment:
This API seems dangerous. Why expose it as uint64_t when in fact it's internally only uint32_t? I think it should be uint32_t all the way. Obviously uint64_t would be better but uint32_t incurs no extra memory so that's the right choice.

Reply:
The main reason is because the lower three bits of the origin Coord can be "stolen" to increase the range of this transient data beyond 32-bits. By stealing these three bits, we can extend the range from 4.3 billion => 34.4 billion. By making this API return 64-bit integers now, we leave room to implement this extension at a later date without needing to change the API.

Reply:
Even if we use the three lower bits of the three 32 bit integers in mOrigin that's only 41 bits! The problem is that the current API suggests that points can be safely encoded (via a cast) as transient data, which is clearly not true. So for now it should be Index32 and if later on we change it to full 64 bit we can change it the API to Index64 since that will introduce an ABI change anyway.

Reply:
(My maths was off there - fixed my above comments).

What you're suggesting is that we ignore those 9 available transient bits because they don't fit neatly into an API? As a user, I would much prefer to get access to those additional bits while knowing that calling `setTransientData(uint64_t)` will only use 41-bits of that uint64_t.

How about we expose an API like the following `setTransientData(uint64_t, bool checkBits=true)` and by default it checks that the upper 23 bits are zero at runtime? If you understand the risks, you can switch this check off for faster performance?

Reply:
As agreed in our last meeting - we will use 32-bit indices as a change to 64-bit could be implemented as an API change only.